import-service/handlers/importFileParser.ts

```diff
@@ -0,0 +1,106 @@
+import { S3Event } from "aws-lambda";
+import {
+  CopyObjectCommand,
+  DeleteObjectCommand,
+  GetObjectCommand,
+  S3Client,
+} from "@aws-sdk/client-s3";
+import { SendMessageCommand, SQSClient } from "@aws-sdk/client-sqs";
+import { Readable, PassThrough } from "stream";
+import csv from "csv-parser";
+import { buildResponse } from "../lib/utils";
+
+export async function handler(event: S3Event) {
+  const bucket = event.Records[0].s3.bucket.name;
+  const key = event.Records[0].s3.object.key;
+  const splitKey = key.split("/");
+  const fileName = splitKey[splitKey.length - 1];
+
+  console.log("bucket: ", bucket, "filePatch: ", key, "file name: ", fileName);
+
+  if (!key.startsWith("uploaded/")) {
+    console.log("Even must by starting in upload folder. Skipping");
+    return;
+  }
+
+  const client = new S3Client({ region: "eu-north-1" });
+  const sqsClient = new SQSClient({ region: "eu-north-1" });
+
+  try {
+    const params = {
+      Bucket: bucket,
+      Key: key,
+    };
+
+    const copyParams = {
+      ...params,
+      Key: `parsed/${fileName}`,
+      CopySource: `${bucket}/${key}`,
+    };
+
+    const getObjectCommand = new GetObjectCommand(params);
+
+    const copyObjectCommand = new CopyObjectCommand(copyParams);
+
+    const deleteObjectCommand = new DeleteObjectCommand(params);
+
+    const file = await client.send(getObjectCommand);
+    console.log({ file });
+    const readStream = file.Body;
+
+    if (!(readStream instanceof Readable)) {
+      throw new Error("Failed to read file");
+    }
+
+    await new Promise((resolve, reject) => {
+      const stream = readStream.pipe(new PassThrough());
+      stream
+        .pipe(csv())
+        .on("data", async (data) => {
+          stream.pause();
+          try {
+            console.log("Send message to SQS", data);
+
+            await sqsClient.send(
+              new SendMessageCommand({
+                QueueUrl:
+                  "https://sqs.eu-north-1.amazonaws.com/298531520651/import-file-batch-queue",
```

Comment:
The preferred way is inject Queue.url by env var (it's more comfortable and flexible) I think